dds_web/web/user.py

```diff
@@ -67,38 +67,35 @@ def index():
 @logging_bind_request
 def confirm_invite(token):
```

Comment:
```
def confirm_invite(token):
    """Confirm invitation."""
    # Verify token and, on success, get row from invite table
    email, invite_row = dds_web.security.auth.verify_invite_token(token)

    # Check the invite exists
    if not invite_row:
        if email and dds_web.utils.email_in_db(email=email):
            flask.flash("Registration has already been completed.")
            return flask.make_response(flask.render_template("user/userexists.html"))
        else:
            flask.flash("This invitation link has expired or is invalid.", "danger")
            return flask.redirect(flask.url_for("auth_blueprint.index"))

    # Save encrypted token to be reused at registration
    # token is in the session already if the user refreshes the page
    if "invite_token" not in flask.session:
        # New visit or session has expired
        flask.session["invite_token"] = token

    form = forms.RegistrationForm()

    # Prefill fields - unit readonly if filled, otherwise disabled
    # These should only be used for display to user and not when actually registering
    # the user, then the values should be fetched from the database again.
    form.unit_name.render_kw = {"disabled": True}
    if invite_row.unit:  # backref to unit
        form.unit_name.data = invite_row.unit.name
        form.unit_name.render_kw = {"readonly": True}

    form.email.data = email
    suggested_username = email.split("@")[0]

    if dds_web.utils.valid_chars_in_username(
        suggested_username
    ) and not dds_web.utils.username_in_db(suggested_username):
        form.username.data = suggested_username

    return flask.render_template("user/register.html", form=form)
```

Reply:
The beginning has changed, added an email check, updated flash text, compiler says use "not in" by following pep guidelines